airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -319,7 +319,10 @@ def parse_response(
         # else -> delegate to record selector
         response_status = self.requester.interpret_response_status(response)
         if response_status.action == ResponseAction.FAIL:
-            error_message = response_status.error_message or f"Request {response.request} failed with response {response}"
+            error_message = (
+                response_status.error_message
+                or f"Request to {response.request.url} failed with response {HttpStream.parse_response_error_message(response)} ({response.status_code})"
```

Comment:
Nit but I think we should either explicitly label the status code just to make things extra clear. In the parentheses it could give the appearance that the `(400)` is part of the response particularly when its at the end of the log message.

Maybe:
`Request to https://blahblah failed with status code 400 and response bad oh bad`

Or if we didn't want to label it, putting it ahead might still be clearer:
`Request to https://blahblah failed with response [400] bad oh bad`